### ISSUE TITLE
All form submissions have to be xhr?  

### ISSUE THREAD

**Issue by prativapokhrel** [Shopify/turbolinks-jwt-sample-app#11]
I have a rails application that is using shopify_app gem, I have also installed webpacker and have setup shopify_app.js. I am not using react/vuejs in the frontend, it's only javascript and jquery. As my app is a multi-page application, I want to know if I need to make all my forms remote: true in order to be able to implement JWT throughout the app?  

I need to implement JWT ASAP in my app, so could someone please help me with this?

**Comment by ko-lem** [comment]
Looks that way

**Comment by overallduka** [comment]
Maybe someone yet needs an answer for this, the answer is YES: all form submissions need to be xhr, and all xhr requests need to include the Authorization header with the sessionToken, something like:

```
  xhr.setRequestHeader("Authorization", "Bearer " + window.sessionToken);
```

**Comment by oliwoodsuk** [comment]
Yeah, we include `remote: true` in all of our form and button helpers.
I thought I'd include something else here too. Whenever we lazy/dynamically load a form, the headers need to be added on manually for it's future requests. below is the Stimulus action I use for this task. 
It's initiated by including this in the form tag:
`data-action: "ajax:beforeSend->form#prepareForm"`



```

import { Controller } from "stimulus"
import Rails from "@rails/ujs"

export default class extends Controller {	
	static targets = ["form"]

	prepareForm(event) {
        const xhr = event.detail[0];
	  if (!xhr.getResponseHeader("ShopOrigin")){
	      xhr.setRequestHeader("Authorization", "Bearer " + window.sessionToken);
	  	  xhr.setRequestHeader("ShopOrigin", window.shopOrigin)
		  document.getElementById('loading-spinner').style.opacity = '0';
	  }
	}


}
```
